src/backend/dx12/src/command.rs

```diff
@@ -1282,12 +1278,54 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
 
                         self.clear_render_target_view(
                             rtv,
-                            cv.into(),
+                            value.into(),
+                            &rect,
+                        );
+                    }
+                }
+                com::AttachmentClear::DepthStencil { depth, stencil } => {
+                    let attachment = {
+                        let dsv_id = sub_pass.depth_stencil_attachment.unwrap();
+                        pass_cache.framebuffer.attachments[dsv_id.0]
+                    };
+
+                    let mut dsv_pool = descriptors_cpu::HeapLinear::new(
+                        &device,
+                        d3d12::D3D12_DESCRIPTOR_HEAP_TYPE_DSV,
+                        clear_rects.len()
+                    );
+
+                    for clear_rect in &clear_rects {
+                        let rect = [get_rect(&clear_rect.rect)];
+
+                        let view_info = device::ViewInfo {
+                            resource: attachment.resource,
+                            kind: attachment.kind,
+                            flags: image::StorageFlags::empty(),
+                            view_kind: image::ViewKind::D2Array,
+                            format: attachment.dxgi_format,
+                            range: image::SubresourceRange {
+                                aspects: if depth.is_some()  { Aspects::DEPTH } else { Aspects::empty() } |
+                                    if stencil.is_some() { Aspects::STENCIL } else {Aspects::empty() },
+                                levels: attachment.mip_levels.0 .. attachment.mip_levels.1,
```

Comment:
AFAICT dx12 only supports having one mip level per view, (`view_render_target_impl` and the dsv version only uses the start of the range)

Reply:
Though still an improvement over hardcoded 0!

Reply:
The same is true for vulkan though, so I think the correct thing to do here is to get the mip level that the attachment is using?

Reply:
The check there, just happening a bit later in `view_image_as_*_impl`.
Technically we wouldn't be here if the user wanted to create a renderpass looking into multiple levels, so an assert is fine.

Reply:
We do assert on that later inside `view_image_as_*_impl`, and technically we shouldn't even come that far if the user wanted to render to multiple levels at once. They'd fail a framebuffer creation.